### PR TITLE
Remove broken multishop support

### PR DESCRIPTION
**Pull request by atomiix**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | The multishop support of this module was never really working and [this fix on the `ObjectModel`](https://github.com/PrestaShop/PrestaShop/pull/25913) made it break even more, leading to an exception. This PR solve the issue by disabling multishop support.
| Type?         | bug fix
| BC breaks?    | yes
| Deprecations? | no
| Fixed ticket? | PrestaShop/PrestaShop#26991, PrestaShop/PrestaShop#26979, PrestaShop/PrestaShop#26980, PrestaShop/PrestaShop#26531 
| How to test?  | Please see related issues

### BC Break
Method `ReassuranceActivity::getAllBlockByLang` has been removed
Method `ReassuranceActivity::getAllBlockByShop` has been removed
The second parameter of the method `ReassuranceActivity::getAllBlockByStatus` has been removed (`$id_shop`)

### CODE REVIEW

File: blockreassurance.php [hunk 1]
```diff
@@ -156,22 +155,21 @@ public function install()
         $sqlQueries[] = 'CREATE TABLE IF NOT EXISTS `' . _DB_PREFIX_ . 'psreassurance_lang` (
             `id_psreassurance` int(10) unsigned NOT NULL,
             `id_lang` int(10) unsigned NOT NULL,
-            `id_shop` int(10) unsigned NOT NULL,
             `title` varchar(255) NOT NULL,
             `description` varchar(255) NOT NULL,
             `link` varchar(255) NOT NULL,
-            PRIMARY KEY (`id_psreassurance`,`id_shop`,`id_lang`)
+            PRIMARY KEY (`id_psreassurance`,`id_lang`)
```
Comment (matks) [inline blockreassurance.php:161]: If we put back the mutishop support later maybe we can keep the SQL structure?

**Review by matks — COMMENTED** [review]
If we disable multishop support we need to inform people in the README https://github.com/PrestaShop/blockreassurance/blob/dev/README.md

**Comment by MatShir** [comment]
> Please give me 1-2 days, I will check how much time would be needed to make it compatible. I will let you know if I can do it. I think it's better to have it working instead of removing it.

The blockreassurance is poorly design for the multistore. Only editing the content of the field text was working, you can't disable/enable an item for a specific context, you can't change the displayed or images. So the configuration part for a specific shop is not working. So I prefer that we assume that the module is not multistore compatible for the configuration rather than do it poorly and do it properly in a second time. 
@kpodemski this PR fixes a lot of major bug. Maybe we can add the multistore compat in another one to avoid to many side effects during test it. WDYT ?

**Comment by kpodemski** [comment]
@MatShir 

Ok

**Comment by atomiix** [comment]
@kpodemski If I remember well, you were willing to add full multishop support to this module, is it still something you plan to do?

**Review by matthieu-rolland — COMMENTED** [review]
LGTM, but the PR needs a rebase and there seems to be a few js lint issues 👍

**Review by PierreRambaud — APPROVED** [review]

**Comment by kpodemski** [comment]
@atomiix 

Yes, it's something I want to do in Q2 with a refacto of the module, so it would be a bigger project

**Review by HanaRebaiQA — CHANGES_REQUESTED** [review]
Hello @atomiix 

I have tested this PR. I found theses issues which are related to this PR : 

_**Check with develop (using the module from dev branch)**_  

**Issue 1 : There is not the possibility to add a new block (A 500 error)** 

1. BO > IMPROVE > Module Manager > blockreassurance > Configure
2. Click on the + icon to add a new Block content
3. Add a new image 
5. See error >> A 500 error is thrown for the first time and there is not the possibility to Save 

image

https://user-images.githubusercontent.com/92912932/159525552-797ee6b4-feee-4956-ba35-6ee34ed4a264.mp4

_**Check with PS1785 (using the module from dev branch)**_  

**Issue 2.  ​All blocks for each shop configured are displayed toghether for every shop in BO & FO**

1. BO > IMPROVE > Module Manager > blockreassurance > Configure
2. Enable multistore and add different shops and group of shops
3. Add a new block with `+` icon for each shop (new shop created, default shop etc)
4. Select a shop
5. See error >> The other blocks already created for the other shops are displayed in the current one (in BO)
image

6. Click on view my shop
7. See error >> The other blocks already created for the other shops are displayed in the current one (in FO)
image

_With PR_

https://user-images.githubusercontent.com/92912932/159520438-ee9526bc-299c-44a4-bf75-4174c24a1514.mp4

_Without PR_ 

https://user-images.githubusercontent.com/92912932/159520903-47a94543-a1cb-4a08-827d-41b8eee6e00d.mp4



**Issue 3. The image icon is not dispayed in BO & FO when adding a new block**

1. BO > IMPROVE > Module Manager > blockreassurance > Configure
3. Add a new block with `+` icon 
4. Add an image for the block and Save
7. See error >> The image icon for the added block is not displayed (In BO)
8. Click on view my shop
9. See error >> The image icon for the added block is not displayed (In FO)

image


_With PR_

https://user-images.githubusercontent.com/92912932/159521855-8b5a7cea-bd77-481e-8549-f4f2843fdc52.mp4

_Without PR_

https://user-images.githubusercontent.com/92912932/159522038-05f5e0d4-01f0-4b65-9105-635e3c34b879.mp4



**Issue 4. I was able to reproduce this issue [#26531](https://github.com/PrestaShop/PrestaShop/issues/26531) again using theses steps** 

1. BO > IMPROVE > Module Manager > blockreassurance > Configure
3. Add a new block with `+` icon 
4. Add all informations and Save
7. Click on edit of the created block
8. Update the language/Redirection type 
10. See error >> The error message `The field "Title" is required at least in your default language ` is displayed

image


https://user-images.githubusercontent.com/92912932/159525814-6b5ba66a-f8b2-46d1-ac8a-9fc67a97666f.mp4


Could you please check ?



Thanks!

**Comment by atomiix** [comment]
hi @HanaRebaiQA, for all the bugs where the other blocks already created for the other shops are displayed in the current one, this is the expected behavior as this PR remove the multishop support. Whatever context you are, blocks are gonna be displayed in every shops.

For the others bugs, I don't understand, I tried on my side but I wasn't able to reproduce it. Can you have a look again?

Here is what I got:

https://user-images.githubusercontent.com/2168836/159734189-4a4de1ea-5a6c-4d48-9723-41909a4bfffe.mov

**Review by HanaRebaiQA — CHANGES_REQUESTED** [review]
Hello @atomiix 

Thank you for your feedback.
Yes, indeed it was a permission problem.

I have those new issues : 
**First issue : In develop branch** 
An error is displayed in FO when preview with the new shop created.


1. Enable multistore and create a new shop
2. BO > Module Manager > Blockreassurance
3. Configure the module with the new shop created
4. Create a block
5. Preview with the new shop in FO 
8. See error > An error is displayed in the whole page

image

https://user-images.githubusercontent.com/92912932/159971682-4280cf55-51c4-4a34-94ab-a2f521f25391.mp4

**Second issue : In 1785 version** 
The image is not displayed with the new shop created in FO after editing the block 

1. Enable multistore and create a new shop
2. BO> Module Manager > Blockreassurance
3. Configure the module with the new shop created
4. Create a block
5. Preview with the new shop in FO 
6. Edit the block with another type (for example: change the Redirection `None` by `CMS Page`
7. Go to FO
8. See error > The image is not displayed again! 

_With PR_ 

https://user-images.githubusercontent.com/92912932/159972785-8d86fd36-b8fd-465d-b8ff-773904a17774.mp4

The type here is `svg`

_Without PR_
image

image

https://user-images.githubusercontent.com/92912932/159973047-28248450-bd6b-4f2c-bf5a-887f56c4474d.mp4


the type here is `svg invisible`

Im waiting for your feedback 

Thank you

**Review by HanaRebaiQA — APPROVED** [review]
Hello @atomiix 

I checked again this PR. Here is my last feedback : 

Check on PS1785 and develop branches : 
1. All issues are verified and fixed : OK 
2. Check with multistore : Ok
3. Check with different languages (RTL and no RTL) in BO & FO : OK
image
4. Check with different image types (svg, jpg, jpe, jpeg, png) : OK
5. Check with different combinations of block Display, Appearance, Content : OK 
6. Check with responsive : OK 
image

7. Check with mobile : OK 
image


https://user-images.githubusercontent.com/92912932/160151898-dc32a9b4-d121-4227-a6f0-68244a8a42c0.mp4


I found other issues but there are not related to this PR.

So, It's QA :heavy_check_mark: 

Thanks!

**Comment by Progi1984** [comment]
Thanks @atomiix & @HanaRebaiQA

**Pull request merged**